iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -599,44 +597,25 @@ private void commitWrite(Table table, String branchName, Long snapshotId, long s
    * @param table             The table we are changing
    * @param startTime         The start time of the commit - used only for logging
    * @param results           The object containing the new files
-   * @param rewritePolicy     The rewrite policy to use for the insert overwrite commit
    * @param partitionSpecId   The table spec_id for partition compaction operation
    * @param partitionPath     The path of the compacted partition
    */
-  private void commitCompaction(Table table, long startTime, FilesForCommit results,
-      RewritePolicy rewritePolicy, Integer partitionSpecId, String partitionPath) {
-    if (rewritePolicy == RewritePolicy.FULL_TABLE) {
-      // Full table compaction
-      Transaction transaction = table.newTransaction();
-      DeleteFiles delete = transaction.newDelete();
-      delete.deleteFromRowFilter(Expressions.alwaysTrue());
-      delete.commit();
-      ReplacePartitions overwrite = transaction.newReplacePartitions();
-      results.dataFiles().forEach(overwrite::addFile);
-      overwrite.commit();
-      transaction.commitTransaction();
-      LOG.debug("Compacted full table with files {}", results);
-    } else {
-      // Single partition compaction
-      List<DataFile> existingDataFiles =
-          IcebergTableUtil.getDataFiles(table, partitionSpecId, partitionPath,
-              partitionPath == null ? Predicate.isEqual(partitionSpecId).negate() : Predicate.isEqual(partitionSpecId));
-      List<DeleteFile> existingDeleteFiles =
-          IcebergTableUtil.getDeleteFiles(table, partitionSpecId, partitionPath,
-              partitionPath == null ? Predicate.isEqual(partitionSpecId).negate() : Predicate.isEqual(partitionSpecId));
-
-      RewriteFiles rewriteFiles = table.newRewrite();
-      rewriteFiles.validateFromSnapshot(table.currentSnapshot().snapshotId());
-
-      existingDataFiles.forEach(rewriteFiles::deleteFile);
-      existingDeleteFiles.forEach(rewriteFiles::deleteFile);
-      results.dataFiles().forEach(rewriteFiles::addFile);
-
-      rewriteFiles.commit();
-      LOG.debug("Compacted partition {} with files {}", partitionPath, results);
-    }
-    LOG.info("Compaction commit took {} ms for table: {} with {} file(s)", System.currentTimeMillis() - startTime,
-        table, results.dataFiles().size());
+  private void commitCompaction(Table table, long startTime, FilesForCommit results, Integer partitionSpecId,
+      String partitionPath) {
+    List<DataFile> existingDataFiles = IcebergTableUtil.getDataFiles(table, partitionSpecId, partitionPath);
+    List<DeleteFile> existingDeleteFiles = IcebergTableUtil.getDeleteFiles(table, partitionSpecId, partitionPath);
+
+    RewriteFiles rewriteFiles = table.newRewrite();
+    rewriteFiles.validateFromSnapshot(getSnapshotId(table, null));
```

Comment:
table is the fresh copy, you should be getting snapshot from outputTable.table

Reply:
Done. But I have a question, why do we pass the fresh copy of the `table` to `commitCompaction` and `commitOverwrite` instead of passing the one from the snapshot?

Reply:
we need a fresh one so we can run the conflict checks on the latest manifest lists, however, we need to specify the proper snapshotid